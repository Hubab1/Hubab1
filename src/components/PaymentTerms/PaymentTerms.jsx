import React, { useEffect, useState, useCallback } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import ArrowBackIos from '@material-ui/icons/ArrowBackIos';
import moment from 'moment';

import API from 'api/api';
import { TOS_TYPE_PAYMENTS } from 'constants/constants';
import { prettyCurrency } from 'utils/misc';

import ActionButton from 'components/ActionButton/ActionButton';
import GenericFormMessage from 'components/GenericFormMessage/GenericFormMessage';
import { Card, H1, LinkButton, ScrollableTermsCardSection, blackLinkRoot, arrowIcon } from 'assets/styles';

const SpacedH1 = styled(H1)`
    margin: 15px 10% 15px 10%;
`;

export const PaymentTerms = ({
    handleClickBack,
    handleTermsAccepted,
    holdingDepositAmount = 0,
    unitNumber,
    communityName,
    leaseStartDate,
    canProceedToPayment,
}) => {
    const [html, setHtml] = useState(undefined);
    const [errors, setErrors] = useState(null);
    const moveInDate = moment(leaseStartDate).format('LL');
    const holdingDepositDisplayedAmount = prettyCurrency(holdingDepositAmount);

    useEffect(() => {
        (async () => {
            try {
                const html = await API.fetchHoldingDepositTerms(canProceedToPayment);
                setHtml(html);
            } catch (e) {
                setErrors([
                    "Oops! we're having troubles finding your Payment and Holiding Deposit Terms. Try again in a bit.",
                ]);
            }
        })();
    }, [canProceedToPayment]);

    const handleContinueClick = useCallback(() => {
        const data = {
            type: TOS_TYPE_PAYMENTS,
            context: {
                time: Date.now(),
                move_in_date: moveInDate,
                holding_deposit: holdingDepositDisplayedAmount,
                community_name: communityName,
                unit_number: unitNumber,
            },
        };

        handleTermsAccepted(data);
    }, [moveInDate, holdingDepositDisplayedAmount, communityName, unitNumber, handleTermsAccepted]);

    return (
        <>
            <SpacedH1>Payment and Holding Deposit Terms</SpacedH1>
            {errors && <GenericFormMessage type="error" messages={errors} />}
            {html && (
                <Card>
                    <ScrollableTermsCardSection>
                        <div
                            dangerouslySetInnerHTML={{
                                __html: html,
                            }}
                        />
                    </ScrollableTermsCardSection>
                </Card>
            )}
            <>
                {canProceedToPayment ? (
                    <>
                        <ActionButton disabled={errors} onClick={handleContinueClick} marginTop={25} marginBottom={20}>
                            Agree and Continue
                        </ActionButton>
                        {!!handleClickBack && (
                            <LinkButton className={blackLinkRoot} onClick={handleClickBack}>
                                <ArrowBackIos classes={{ root: arrowIcon }} /> Go Back
                            </LinkButton>
                        )}
                    </>
                ) : (
                    !!handleClickBack && (
                        <ActionButton onClick={handleClickBack} marginTop={25} marginBottom={20}>
                            Go Back
                        </ActionButton>
                    )
                )}
            </>
        </>
    );
};

PaymentTerms.propTypes = {
    handleClickBack: PropTypes.func.isRequired,
    handleTermsAccepted: PropTypes.func,
    holdingDepositAmount: PropTypes.number.isRequired,
    unitNumber: PropTypes.string.isRequired,
    communityName: PropTypes.string.isRequired,
    leaseStartDate: PropTypes.string.isRequired,
    canProceedToPayment: PropTypes.bool.isRequired,
};

export default PaymentTerms;
