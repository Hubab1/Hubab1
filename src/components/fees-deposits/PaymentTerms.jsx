import React, {Fragment} from 'react';
import { css } from 'emotion';
import styled from '@emotion/styled';
import ArrowBackIos from '@material-ui/icons/ArrowBackIos'

import ActionButton from 'components/common/ActionButton/ActionButton';
import GreenCheckIcon from 'components/common/GreenCheckIcon';
import { Card, P, H1, LinkButton, ScrollableTermsCardSection, blackLinkRoot, arrowIcon } from 'assets/styles';

const SpacedH1 = styled(H1)`
    margin: 15px 10% 15px 10%;
`

const bulletBox = css`
    margin-bottom: 30px
`

const BulletP = styled(P)`
    font-size: 15px
`

const BodyP = styled(P)`
    font-size: 14px;
    margin: 0 0 12px 0;
`

// TODO: Make this consistent with TermsPage once we get designs
export const PaymentTerms = ({handleClickBack, goToPayment}) => {
    return <Fragment>
        <SpacedH1>Application Fees and Holding Deposit</SpacedH1>
        <Card>
            <ScrollableTermsCardSection>
                {/*
                    removing these for now, since the mocks removed them
                <div className={bulletBox}>
                    <BulletP>
                        <GreenCheckIcon/>
                        {' '}
                        Applied to the security deposit
                    </BulletP>
                    <BulletP>
                        <GreenCheckIcon/>{' '}
                        Refundable if cancelled within 24 hours
                    </BulletP>
                    <BulletP>
                        <GreenCheckIcon/>
                        {' '}
                        Refundable if denied
                    </BulletP>
                </div>*/}
                <BodyP>Applicant has deposited, with owner’s agent a holding deposit as specified below to secure the rental of the below described residential unit, subject to the following conditions:</BodyP>
                <BodyP>1) If accepted, the holding deposit shall be applied to the security deposit due at move-in.</BodyP>
                <BodyP>2) Applicant shall have 24hours following the date of this application to withdraw the application and receive fullrefund  of  said  holding  deposit.  (If  applicant  cancels  rental  after  24  hours  and  was  otherwise  approved,  theholding deposit is forfeited.)</BodyP>
                <BodyP>3) Holding deposit shall be fully refunded (not application fee) in case of denial of application.</BodyP>
                <BodyP>4) Applicant agrees to take financial responsibility of the premises onMay1,2018and pay the balance of rentdue and security deposits in full, on that date.</BodyP>
                <BodyP>5) Applicant  agrees  to  provide  proof  of$100,000.00liability  insurance  by  lease  commencement  date.  Thepreferred provider is: erenterplan.com (888) 205-8118.6)A non-refundable application processing fee is required to prepare the rental agreement documents and toverify applicant’s credit history, rental reference and employment.7)If said premises are not vacated by present resident on proposed move-out date and the present resident isstill  in  possession,  the  holding  deposit  will  be  returned  in  full  to  applicant.  The  holding  deposit  does  notguarantee occupancy.8)Credit  approval  is  based  on:  (1)  one  year  positive  rental  history;  (2)  good  credit  (note:  late  mortgagepayments, foreclosure, or bankruptcy may result in denial); (3) one year current employment history</BodyP>
            </ScrollableTermsCardSection>
        </Card>
        { !!handleClickBack && 
            <Fragment>
                <ActionButton onClick={goToPayment} marginTop={25} marginBottom={20}>
                    Agree and Continue
                </ActionButton>
                <LinkButton className={blackLinkRoot} onClick={handleClickBack}>
                    <ArrowBackIos classes={{root: arrowIcon}}/> Go Back
                </LinkButton>
            </Fragment>
        }
    </Fragment>
}

export default PaymentTerms;