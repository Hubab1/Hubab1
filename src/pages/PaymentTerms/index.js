import PaymentTermsPage from './PaymentTermsPage';

export default PaymentTermsPage;
