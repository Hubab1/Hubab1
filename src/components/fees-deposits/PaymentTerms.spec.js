import React from 'react';
import { shallow } from 'enzyme';

import { PaymentTerms } from './PaymentTerms';

let defaultProps;
beforeEach(() => {
    defaultProps = {
        handleClickBack: jest.fn(),
        goToPayment: jest.fn(),
        communityName: 'Monterey Pines Apartments',
        holdingDepositAmount: 500,
        leaseStartDate: '2020-10-30',
        unitNumber: '24',
    };
});

it('calls handleClickBack when Go Back is clicked', () => {
    const wrapper = shallow(<PaymentTerms {...defaultProps} />);
    expect(wrapper.getElement()).toMatchSnapshot();
});

it('calls handleClickBack when Go Back is clicked', () => {
    const wrapper = shallow(<PaymentTerms {...defaultProps} />);

    wrapper.find('LinkButton').simulate('click');
    expect(defaultProps.handleClickBack).toHaveBeenCalled();
});

it('calls goToPayment when Go Back is clicked', () => {
    const wrapper = shallow(<PaymentTerms {...defaultProps} />);

    wrapper.find('ActionButton').simulate('click');
    expect(defaultProps.goToPayment).toHaveBeenCalled();
});
