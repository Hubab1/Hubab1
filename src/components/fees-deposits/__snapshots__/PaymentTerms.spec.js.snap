// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls handleClickBack when Go Back is clicked 1`] = `
<React.Fragment>
  <ForwardRef(render)>
    Application Fees and Holding Deposit
  </ForwardRef(render)>
  <ForwardRef(render)>
    <ForwardRef(render)>
      <ForwardRef(render)>
        Applicants will deposit with owner’s agent a holding deposit in the amount of 
        $500
         (“Holding Deposit”) for the residential unit 
        Monterey Pines Apartments
         Unit 
        24
         (“Unit”), subject to the following conditions:
      </ForwardRef(render)>
      <ForwardRef(render)>
        <b>
          1)
        </b>
         If accepted, the Holding Deposit shall be applied to the security deposit due at move-in
      </ForwardRef(render)>
      <ForwardRef(render)>
        <b>
          2)
        </b>
         Applicants shall have 
        <b>
          <u>
            24 hours
          </u>
        </b>
         following the date of this application to withdraw the application and receive full refund of said Holding Deposit. If applicants cancel rental after 24 hours and was otherwise approved, the holding deposit is forfeited.
      </ForwardRef(render)>
      <ForwardRef(render)>
        <b>
          3)
        </b>
         Holding Deposit shall be fully refunded (not application fee) in case of denial of application.
      </ForwardRef(render)>
      <ForwardRef(render)>
        <b>
          4)
        </b>
         Applicants agree to take financial responsibility of the premises on 
        <b>
          <u>
            October 29, 2020
          </u>
        </b>
         and pay the balance of rent and security deposits due on that date.
      </ForwardRef(render)>
      <ForwardRef(render)>
        <b>
          5)
        </b>
         Applicants agree to provide proof of liability insurance and proof of Utilities start services, as required by the lease if applicable, by lease commencement date.
      </ForwardRef(render)>
      <ForwardRef(render)>
        <b>
          6)
        </b>
         If the Unit is not vacated by present resident on proposed move-out date and the present resident is still in possession, the Holding Deposit will be returned in full to applicants. The holding deposit does not guarantee occupancy.
      </ForwardRef(render)>
      <ForwardRef(render)>
        <b>
          7)
        </b>
         The Primary Applicant shall be responsible for depositing with owner’s Agent the Holding Deposit. The Holding Deposit is applicable to all Applicants jointly. Any refund due will be made payable the Primary Applicant and it shall be the responsibility of all Applicants to work out between themselves the manner of dividing the Holding Deposit, if any.
      </ForwardRef(render)>
      <ForwardRef(render)>
        <b>
          8)
        </b>
         By clicking “Agree and Continue”, Applicants agree to the above terms and conditions for the Holding Deposit.
      </ForwardRef(render)>
    </ForwardRef(render)>
  </ForwardRef(render)>
  <React.Fragment>
    <ActionButton
      color="primary"
      marginBottom={20}
      marginTop={25}
      onClick={[MockFunction]}
      successGreen={false}
      variant="contained"
    >
      Agree and Continue
    </ActionButton>
    <ForwardRef(render)
      className="css-vq6rzm"
      onClick={[MockFunction]}
    >
      <UNDEFINED
        classes={
          Object {
            "root": "css-13z8uwt",
          }
        }
      />
       Go Back
    </ForwardRef(render)>
  </React.Fragment>
</React.Fragment>
`;
